utils/replicaset/canary.go

```diff
@@ -83,7 +83,15 @@ func DesiredReplicaCountsForCanary(rollout *v1alpha1.Rollout, newRS, stableRS *a
 // For more examples, check the TestCalculateReplicaCountsForCanary test in canary/canary_test.go
 func CalculateReplicaCountsForCanary(rollout *v1alpha1.Rollout, newRS *appsv1.ReplicaSet, stableRS *appsv1.ReplicaSet, oldRSs []*appsv1.ReplicaSet) (int32, int32) {
 	rolloutSpecReplica := defaults.GetReplicasOrDefault(rollout.Spec.Replicas)
-	setWeight := GetCurrentSetWeight(rollout)
+	setWeight := int32(0)
+	if scs, use := UseSetCanaryScale(rollout); use == true {
+		if scs.Replicas != nil {
+			return *scs.Replicas, rolloutSpecReplica
+		}
+		setWeight = *scs.Weight
```

Comment:
I feel there is potential for nil dereference. I would suggest to a little more defensive and do something like:

```
		if scs.Replicas != nil {
			return *scs.Replicas, rolloutSpecReplica
		} else if scs.Weight != nil {
			setWeight = *scs.Weight
		} else {
			setWeight = GetCurrentSetWeight(rollout)
		} 
```